dockhub.py

```diff
@@ -41,21 +41,21 @@ def die(msg):
 @handle_http_errors
 def get_auth_token():
     docker_api_url = f'{DOCKERHUB_URL}/users/login/'
-    username = getenv('DH_USERNAME')
-    if username is not None:
-        password = getenv('DH_PASSWORD')
-        if password is not None:
-            auth = {"username": username, "password": password}
-            auth_request = requests.post(docker_api_url, headers=CONTENT_HEADER, json=auth)
-            if auth_request.status_code == 200:
-                return auth_request.json()['token']
-            else:
-                die('Non-200 response received from DockerHub. Verify your credentials and try again')
-        else:
-            die('For security, you must set your password as the variable DH_PASSWORD in your ENV')
-    else:
+    username = getenv('DH_USERNAME', '').strip()
+    if not username:
```

Comment:
I inverted some of these if blocks so that they "drop out early". This sometimes makes things easier to read because you don't have to keep as much context in your head as you're reading and it can reduce indentation levels.

Also, I changed `getenv` to take an empty string if the variable isn't there. That way it's always returning a string and you can call `strip()` on it and then check if it's falsey which covers when it's not there as well as when it's the empty string.

Reply:
Like it